test/client-unref.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const { test } = require('tap')
+const { Client } = require('..')
+const http = require('http')
+
+test('end process on idle', (t) => {
+  t.plan(2)
+
+  const server = http.createServer((req, res) => {
+    res.end()
+  })
+  t.teardown(server.close.bind(server))
```

Comment:
you need to remove this and unref() the server instead

Reply:
Tried that as well but it didn't help.

Reply:
☹️